ansible/roles/test/files/ptftests/radv_ipv6_ra_test.py

```diff
@@ -0,0 +1,240 @@
+                                                src_ip=self.downlink_vlan_ip6,
+                                                dst_ip=ALL_NODES_IPV6_MULTICAST_ADDRESS)
+        self.masked_rapkt = self.mask_off_dont_care_ra_packet_fields(rapkt)
+
+
+    def tearDown(self):
+        DataplaneBaseTest.tearDown(self)
+
+    """
+    @summary: Verify received RA packet
+
+    """
+
+    def verify_periodic_router_advertisement(self):
+        testutils.verify_packet(
+                                self,
+                                self.masked_rapkt,
+                                self.ptf_port_index,
+                                self.radv_max_ra_interval+1)
+        logging.info("Received unsolicited RA from:%s on PTF eth%d",
+                                                            self.downlink_vlan_ip6,
+                                                            self.ptf_port_index)
+
+
+    def runTest(self):
+        count = 5
+        while count > 0:
+            self.verify_periodic_router_advertisement()
+            count = count - 1
+
+"""
+@summary: This test validates the solicited router advertisement sent on the VLAN network of the ToR
+We simulate the ToR sending the ICMPv6 router solicitation packet through one of the PTF port (eg eth0)
+
+"""
+
+class RadvSolicitedRATest(DataplaneBaseTest):
+    def __init__(self):
+        DataplaneBaseTest.__init__(self)
+
+    def setUp(self):
+        DataplaneBaseTest.setUp(self)
+
+        self.test_params = testutils.test_params_get()
+
+        self.hostname = self.test_params['hostname']
+        self.downlink_vlan_mac = self.test_params['downlink_vlan_mac']
+        self.downlink_vlan_ip6 = self.test_params['downlink_vlan_ip6']
+        self.ptf_port_index = int(self.test_params['ptf_port_index'])
+        self.ptf_port_mac = self.dataplane.get_mac(0, self.ptf_port_index)
+        self.ptf_port_ip6 = self.test_params['ptf_port_ip6']
+        self.radv_max_ra_interval = int(self.test_params['max_ra_interval'])
+
+        rapkt = self.create_icmpv6_router_advertisement_packet(
+                                    src_mac=self.downlink_vlan_mac,
+                                    dst_mac=self.ptf_port_mac,
+                                    src_ip=self.downlink_vlan_ip6,
+                                    dst_ip=self.ptf_port_ip6)
+
+        self.masked_rapkt = self.mask_off_dont_care_ra_packet_fields(rapkt)
+        self.rs_packet = self.create_icmpv6_router_solicitation_packet()
+
+    def tearDown(self):
+        DataplaneBaseTest.tearDown(self)
+
+
+    """
+    @summary: Creates a solicited RA packet originating from PTF port
+
+    """
+    def create_icmpv6_router_solicitation_packet(self):
+        ether = scapy.layers.l2.Ether(
+                            dst=ALL_ROUTERS_MULTICAST_MAC_ADDRESS,
+                            src=self.ptf_port_mac)
+        ip6 = scapy.layers.inet6.IPv6(
```

Comment:
Suggest placing first argument on this line and aligning arguments on following lines with it.